Files/Filesystem/StorageItems/FtpStorageFile.cs

```diff
@@ -0,0 +1,273 @@
+        
+        public IAsyncOperation<StorageStreamTransaction> OpenTransactedWriteAsync() => throw new NotSupportedException();
+        
+        public IAsyncOperation<StorageFile> CopyAsync(IStorageFolder destinationFolder)
+        {
+            return CopyAsync(destinationFolder, Name, NameCollisionOption.FailIfExists);
+        }
+        
+        public IAsyncOperation<StorageFile> CopyAsync(IStorageFolder destinationFolder, string desiredNewName)
+        {
+            return CopyAsync(destinationFolder, desiredNewName, NameCollisionOption.FailIfExists);
+        }
+        
+        public IAsyncOperation<StorageFile> CopyAsync(IStorageFolder destinationFolder, string desiredNewName, NameCollisionOption option)
+        {
+            return AsyncInfo.Run(async (cancellationToken) =>
+            {
+                var ftpClient = _viewModel.GetFtpInstance();
+
+                if (!await ftpClient.EnsureConnectedAsync())
+                {
+                    return null;
+                }
+
+                var createOption = option switch
+                {
+                    NameCollisionOption.FailIfExists => CreationCollisionOption.FailIfExists,
+                    NameCollisionOption.GenerateUniqueName => CreationCollisionOption.GenerateUniqueName,
+                    NameCollisionOption.ReplaceExisting => CreationCollisionOption.ReplaceExisting,
+                    _ => CreationCollisionOption.FailIfExists
+                };
+
+                var file = await destinationFolder.CreateFileAsync(desiredNewName, createOption);
+                var stream = await file.OpenStreamForWriteAsync();
+
+                if (await ftpClient.DownloadAsync(stream, FtpPath, token: cancellationToken))
+                {
+                    return file;
+                }
+
+                return null;
+            });
+        }
+        
+        public IAsyncAction CopyAndReplaceAsync(IStorageFile fileToReplace)
+        {
+            return AsyncInfo.Run(async (cancellationToken) =>
+            {
+                var ftpClient = _viewModel.GetFtpInstance();
+
+                if (!await ftpClient.EnsureConnectedAsync())
+                {
+                    return;
+                }
+
+                var stream = await fileToReplace.OpenStreamForWriteAsync();
+                await ftpClient.DownloadAsync(stream, FtpPath, token: cancellationToken);
+            });
+        }
+        
+        public IAsyncAction MoveAsync(IStorageFolder destinationFolder)
+        {
+            return MoveAsync(destinationFolder, Name, NameCollisionOption.FailIfExists);
+        }
+
+        public IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName)
+        {
+            return MoveAsync(destinationFolder, desiredNewName, NameCollisionOption.FailIfExists);
+        }
+
+        public IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName, NameCollisionOption option)
```

Comment:
There're some redundant implementations in `FtpStorageFile`, I don't think we should support moving files from/to a FTP server. Will cleanup later.